go-controller/pkg/annotation/node.go

```diff
@@ -0,0 +1,32 @@
+package annotation
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func EnsureNetworkID(watchFactory factory.NodeWatchFactory, nInfo util.NetInfo, networkID *int) error {
+	if networkID == nil {
+		return fmt.Errorf("nil network id pointer")
+	}
+	if *networkID != util.InvalidNetworkID {
+		return nil
+	}
+	nodes, err := watchFactory.GetNodes()
+	if err != nil {
+		return fmt.Errorf("failed to get nodes: %v", err)
+	}
+	for _, node := range nodes {
+		*networkID, err = util.ParseNetworkIDAnnotation(node, nInfo.GetNetworkName())
+		if err != nil {
+			//TODO Warning
```

Comment:
let's have this properly baked in from the beginning. Is it a log or an even that is missing ? 

Is there something we wouldn't know how to do regarding this ?

Reply:
We should continue just in "not found" and return error back on error, but there is no proper way at base code to differenciate, I will try to make the specific error public so we can handle it.

Reply:
There is a `util.IsAnnotationNotSetError` we can use.